RosNet/DataModel/RosBag.cs

```diff
@@ -43,6 +45,53 @@ public override string ToString()
             return s;
         }
 
+        /// <summary>
+        /// Makes a dictionary of all the topics and fieldnames in rosbag
+        /// </summary>
+        /// <returns>Dictionary with topics and corresponding fieldnames</returns>
+        public Dictionary<string, List<string>> GetConnectionFields()
+        {
+            var fieldsByTopic = new Dictionary<string, List<string>>();
+            foreach (Connection conn in Connections.Values)
+            {
+                fieldsByTopic.Add(conn.Topic, new List<string>());
+                foreach (FieldValue field in conn.MessageDefinition)
+                {
+                    fieldsByTopic[conn.Topic].Add(field.Name);
+                }
+            }
+            return fieldsByTopic;
+        }
 
+        /// <summary>
+        /// Makes a dictionary of timestamps and fieldvalues of field with fieldname fieldName
+        /// </summary>
+        /// <returns>Dictionary with time and corresponding fieldvalue</returns>
+        public Dictionary<Time, FieldValue> GetTimeSeries(string topic, string fieldName)
+        {
+            var timeSeries = new Dictionary<Time, FieldValue>();
+
+            //todo:finn bedre løsning
+            foreach (Connection conn in Connections.Values)
```

Comment:
Using dictionaries, you can access a key-value in O(1), instead of this nested for loop. I suggest a ConnectionsByTopic and MessagesByFieldName.

Reply:
Yes, that was something like what we thought. For now we just put down this (bad) solution, but we can try to do it that way later today or tomorrow

Reply:
So you suggest making the whole "Connections" dict be based on Topic instead of UID? I believe ROS has no guarantee for unique Topics, which is my main concern

Reply:
Added issue #78 to look at this later